### PR TITLE
PatrickLonga patch

### PR DESCRIPTION
**Pull request by patricklonga**
A first pass on the draft.

### CODE REVIEW

File: spec/draft-barnes-cfrg-4q.md [hunk 1]
```diff
@@ -109,29 +112,35 @@ N = 0x29cbc14e5e0a72f05397829cbc14e5dfbd004dfe0f79992fb2540ec7768ce7
 
 # Curve Points
 
-Elements of GF(p) are represented as 16-octet little-endian integers.  An element
-x0 + x1*i of GF(p^2) is represented on the wire by the concatenation of the byte strings for
-x0 and x1. Implementations will use whatever internal representation they desire, but we will
-describe the operations on elements of GF(p^2) assuming that X=x0+x1*i, with each x0 and x1
-an element of GF(p).
+Elements a = (a0, a1, ... , a126)_2 in GF(p) are represented as 128-bit (16-octet) little-endian 
+integers by concatenating a || 0, i.e., elements of GF(p) are encoded as (a0, a1, ... , a126, 0)_2.
```
Comment (wbl) [inline spec/draft-barnes-cfrg-4q.md]: I find this line confuses the heck out of me. Which bit in a byte is the MSB, which one the LSB? Are there any devices where octets aren't 8 bits?

File: spec/draft-barnes-cfrg-4q.md [hunk 2]
```diff
@@ -78,11 +78,12 @@ even more speedups.
 * Faster due to endomorphisms
 * More magic: potentially risky.
 
-# Four-dimensional decompsitions
+# Four-dimensional decompositions
```
Comment (bifurcation) [inline spec/draft-barnes-cfrg-4q.md]: Let's just delete this heading and merge with the previous section

File: spec/draft-barnes-cfrg-4q.md [hunk 3]
```diff
@@ -94,13 +95,15 @@ Curve4Q is the twisted Edwards curve over GF(p^2) defined by the following curve
 equation:
 
 ~~~~~
-Y^2 - X^2 = 1 + d * X^2 * Y^2, with
+E: -x^2 + y^2 = 1 + d * x^2 * y^2, with
 
-d = 0x00000000000000e40000000000000142 +
-    0x5e472f846657e0fcb3821488f1fc0c8d * i
+d = 0x00000000000000e40000000000000142 + 0x5e472f846657e0fcb3821488f1fc0c8d * i
```
Comment (bifurcation) [inline spec/draft-barnes-cfrg-4q.md]: Unfortunately, this wrapping is necessary in order to be compatible with the 72-column limit for RFCs.  ¯_(ツ)_/¯

File: spec/draft-barnes-cfrg-4q.md (hunk 1 above)
Comment (bifurcation) [inline spec/draft-barnes-cfrg-4q.md]: Well, an "octet" is by definition 8 bits :)  And I wouldn't worry about platforms where a byte isn't 8 bits.

That said, I agree that this is not a very useful encoding.  Changing byte-level endianness is common; changing bit-level endianness is a huge pain.  I think people understand that "little-endian integer" means, but if we want to be more explicit here, we could just steal from RFC 7748:

```
   The u-coordinates are elements of the underlying field GF(2^255 - 19)
   or GF(2^448 - 2^224 - 1) and are encoded as an array of bytes, u, in
   little-endian order such that u[0] + 256*u[1] + 256^2*u[2] + ... +
   256^(n-1)*u[n-1] is congruent to the value modulo p and u[n-1] is
   minimal.
```

Comment (patricklonga) [inline spec/draft-barnes-cfrg-4q.md]: Agreed. Please, use whatever notation you think is appropriate here. Note that I was trying to follow the EdDSA draft (draft-irtf-cfrg-edsa-05), where they give explicit details about conversion bit string <-> octet string and their little endian encoding (see their Section 2).  
Anyway, the important issue that I wanted to highlight is that we need to specify that the most significant bit of the 16th octet of a field element encoding is cleared.

**Pull request merged**